FP `unused_extern_crates`

<!--
Thank you for filing a bug report! 🐛 Please provide a short summary of the bug,
along with any information you feel relevant to replicating the bug.
-->

I tried this code:
(edition 2024)
```rust
#![warn(unused_extern_crates)]

#![no_implicit_prelude]
extern crate std;
fn main() {
    let r = 1u16..10;
    std::println!("{:?}", r.is_empty());
}

```
=>

```
warning: `extern crate` is not idiomatic in the new edition
 --> src/main.rs:4:1
  |
4 | extern crate std;
  | ^^^^^^^^^^^^^^^^^
  |
note: the lint level is defined here
 --> src/main.rs:1:9
  |
1 | #![warn(unused_extern_crates)]
  |         ^^^^^^^^^^^^^^^^^^^^
help: convert it to a `use`
  |
4 - extern crate std;
4 + use std;
```


the suggested code

```rust
#![warn(unused_extern_crates)]

#![no_implicit_prelude]
use std;
fn main() {
    let r = 1u16..10;
    std::println!("{:?}", r.is_empty());
}

```

does not compile:
```
error[E0432]: unresolved import `std`
 --> src/main.rs:4:5
  |
4 | use std;
  |     ^^^ no external crate `std`

For more information about this error, try `rustc --explain E0432`.
```

--- a/compiler/rustc_resolve/src/check_unused.rs
+++ b/compiler/rustc_resolve/src/check_unused.rs
@@ -193,6 +193,16 @@ impl<'a, 'ra, 'tcx> UnusedImportCheckVisitor<'a, 'ra, 'tcx> {
                 continue;
             }
 
+            let module = self
+                .r
+                .get_nearest_non_block_module(self.r.local_def_id(extern_crate.id).to_def_id());
+            if module.no_implicit_prelude {
+                // If the module has `no_implicit_prelude`, then we don't suggest
+                // replacing the extern crate with a use, as it would not be
+                // inserted into the prelude. User writes `extern` style deliberately.
+                continue;
+            }
+
             let vis_span = extern_crate
                 .vis_span
                 .find_ancestor_inside(extern_crate.span)

--- a/tests/ui/resolve/extern-crate-lint-issue-141785.rs
+++ b/tests/ui/resolve/extern-crate-lint-issue-141785.rs
@@ -1,0 +1,11 @@
+//@ check-pass
+//@ edition:2018
+
+#![no_implicit_prelude]
+#![warn(unused_extern_crates)]
+
+extern crate std;
+fn main() {
+    let r = 1u16..10;
+    std::println!("{:?}", r.is_empty());
+}

